joblib/hashing.py

```diff
@@ -30,7 +30,7 @@ class _ConsistentSet(object):
         whatever the order of its items.
     """
     def __init__(self, set_sequence):
-        self._sequence = sorted(set_sequence)
+        self._sequence = sorted(set_sequence, key=hash)
```

Comment:
If I understand Python 3.3 changelog, this should not be deterministic anymore in recent versions of Python:

https://docs.python.org/3/whatsnew/3.3.html#builtin-functions-and-types